libs/store/src/lib/reducers/add.ts

```diff
@@ -0,0 +1,83 @@
+import { AppStateModel, AppTournamentId } from '@razor/models';
+import { extractId, extractIdType } from '@razor/util';
+import {
+  addPlayerReducerPayload,
+  addRaceReducerPayload,
+  addTournamentReducerPayload,
+} from '../payloads';
+
+// Basic Add Operations
+export const addTournamentReducer = (
+  state: AppStateModel,
+  payload: addTournamentReducerPayload,
+): AppStateModel => {
+  const { tournamentId, tournament } = payload;
+  const newState: AppStateModel = {
+    ...state,
+    tournamentsModel: {
+      ...state.tournamentsModel,
+      [tournamentId]: tournament,
+    },
+  };
+  return newState;
+};
+
+export const addRaceReducer = (
+  state: AppStateModel,
+  payload: addRaceReducerPayload,
+): AppStateModel => {
+  const { raceId, race } = payload;
+  const tournamentId: AppTournamentId = extractId(
+    raceId,
+    extractIdType.race,
+    extractIdType.tournament,
+  ) as AppTournamentId;
+
+  if (raceId) {
+    const newState: AppStateModel = {
+      ...state,
+      tournamentsModel: {
+        ...state.tournamentsModel,
+        [tournamentId]: {
+          ...state.tournamentsModel[tournamentId],
+          raceIds: [...state.tournamentsModel[tournamentId].raceIds, raceId],
+        },
+      },
+      racesModel: {
+        ...state.racesModel,
+        [raceId]: { ...race },
+      },
+    };
+    return newState;
+  } else {
+    return state;
+  }
+};
+
+export const addPlayerReducer = (
+  state: AppStateModel,
+  payload: addPlayerReducerPayload,
+): AppStateModel => {
+  const { tournamentId, playerId, player } = payload;
+  const newState: AppStateModel = {
+    ...state,
+    tournamentsModel: {
+      ...state.tournamentsModel,
+      [tournamentId]: {
+        ...state.tournamentsModel[tournamentId],
+        playerIds: [
+          ...(state.tournamentsModel[tournamentId] &&
```

Comment:
You can simplify this statement as below:
`state.tournamentsModel[tournamentId]?.playerIds || []`

Reply:
This logic is no longer used because functions are checking tournament (or any other related value) availability at the top of the function. 39ef350e9fa663424ef9b78c7f25a8c240263c36